staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4131,8 +4134,86 @@ type Event struct {
 	// Type of this event (Normal, Warning), new types could be added in the future
 	// +optional
 	Type string `json:"type,omitempty" protobuf:"bytes,9,opt,name=type"`
+
+	// Time when this Event was first observed.
+	// +optional
+	EventTime metav1.MicroTime `json:"eventTime,omitempty" protobuf:"bytes,10,opt,name=eventTime"`
```

Comment:
The "normal" way is to add a new GroupVersion (we have that with events.k8s.io/v1beta1) with the extended information and use annotations in the old types to store additional info the types cannot express yet.